docs/code/dns/main.c

```diff
@@ -0,0 +1,80 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <uv.h>
+
+uv_loop_t *loop;
+
+void alloc_buffer(uv_handle_t *handle, size_t suggested_size, uv_buf_t *buf) {
+  buf->base = malloc(suggested_size);
+  buf->len = suggested_size;
+}
```

Comment:
> Just that the uvbook is already out there. Instead of reviewing the entire book, we could take it as a known good commodity, and any necessary fixes could be made as they are discovered. That's the kind of thing that is friendly to first time contributors too.

Sure, that's what this PR is about! :-)